processor/filterprocessor/filter_processor_test.go

```diff
@@ -184,20 +189,103 @@ func TestFilterMetricProcessor(t *testing.T) {
 				}
 			}
 
-			cErr := fmp.ConsumeMetricsData(context.Background(), md)
+			cErr := fmp.ConsumeMetrics(context.Background(), pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{md}))
 			assert.Nil(t, cErr)
 
-			gotMetrics := next.AllMetrics()[0].Metrics
-			require.Equal(t, len(test.outMN), len(gotMetrics))
-			for idx, out := range gotMetrics {
+			got := next.AllMetrics()
+			require.Equal(t, 1, len(got))
+			gotMD := pdatautil.MetricsToMetricsData(got[0])
+			require.Equal(t, 3, len(gotMD))
+			assert.EqualValues(t, consumerdata.MetricsData{}, gotMD[0])
+			require.Equal(t, len(test.outMN), len(gotMD[1].Metrics))
+			for idx, out := range gotMD[1].Metrics {
 				assert.Equal(t, test.outMN[idx], out.MetricDescriptor.Name)
 			}
-
+			assert.EqualValues(t, consumerdata.MetricsData{Metrics: []*metricspb.Metric{}}, gotMD[2])
 			assert.NoError(t, fmp.Shutdown(ctx))
 		})
 	}
 }
 
+func TestFilterMetricProcessorEmptyNilMetrics(t *testing.T) {
+	// next stores the results of the filter metric processor
+	next := &etest.SinkMetricsExporter{}
+	cfg := &Config{
+		ProcessorSettings: configmodels.ProcessorSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+		Metrics: MetricFilters{
+			Include: regexpMetricsFilterProperties,
+			Exclude: &filtermetric.MatchProperties{
+				Config: filterset.Config{
+					MatchType: filterset.Strict,
+				},
+				MetricNames: []string{
+					"prefix_test_match",
+					"test_contains_match",
+				},
+			},
+		},
+	}
+	fmp, err := newFilterMetricProcessor(next, cfg)
+	assert.NotNil(t, fmp)
+	assert.Nil(t, err)
+
+	caps := fmp.GetCapabilities()
+	assert.Equal(t, false, caps.MutatesConsumedData)
+	ctx := context.Background()
+	assert.NoError(t, fmp.Start(ctx, nil))
+
+	md := consumerdata.MetricsData{
+		Metrics: make([]*metricspb.Metric, len(inMetricNames)),
+	}
+
+	for idx, in := range inMetricNames {
+		md.Metrics[idx] = &metricspb.Metric{
+			MetricDescriptor: &metricspb.MetricDescriptor{
+				Name: in,
+			},
+		}
+	}
+
+	cErr := fmp.ConsumeMetrics(
+		context.Background(),
+		pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
+			{},
+			md,
+			{
+				Metrics: []*metricspb.Metric{},
+			},
+		}))
```

Comment:
Shouldn't this just be:

```suggestion
		pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
			{},
			{
				Metrics: []*metricspb.Metric{},
			},
		}))
```

And then you can delete ~70% of the rest of the code from this test?

Reply:
(since you already cover testing the case with data above)

Reply:
not really, because if I stop at a nil / empty and I don't check for the second entry that has data I have a problem. That's why initially I added in the previous test.

Reply:
Oh okay, maybe it would be better to just put that as a comment in the previous test so its clear why this is done, and delete this test then 😄

Reply:
Done, made the test to accept a list as input/output